### PR TITLE
Add a flag for Non-GCP mode NEG controller.

### PR DESCRIPTION
**Pull request by cxhiano**
With '--enable-non-gcp-mode', NEG controller will create NEGs of type NON_GCP_PRIVATE_IP_PORT instead of GCE_VM_IP_PORT.

### CODE REVIEW

File: pkg/neg/manager.go [hunk 1]
```diff
@@ -113,9 +113,15 @@ func (manager *syncerManager) EnsureSyncers(namespace, name string, newPorts neg
 		syncerKey := getSyncerKey(namespace, name, svcPort, portInfo)
 		syncer, ok := manager.syncerMap[syncerKey]
 		if !ok {
+			networkEndpointType := negtypes.VMNetworkEndpointType
```
Comment (cxhiano) [inline pkg/neg/manager.go]: type negtypes.NEGType string

const (
VMNetworkEndpiontType = NEGType(VM)
)

File: pkg/neg/syncers/transaction.go [hunk 2]
```diff
@@ -35,7 +35,8 @@ import (
 type transactionSyncer struct {
 	// metadata
 	negtypes.NegSyncerKey
-	negName string
+	negName             string
+	networkEndpointType string
```
Comment (cxhiano) [inline pkg/neg/syncers/transaction.go]: add comment

File: pkg/neg/syncers/utils.go [hunk 3]
```diff
@@ -286,6 +298,11 @@ func makeEndpointBatch(endpoints negtypes.NetworkEndpointSet) (map[negtypes.Netw
 			IpAddress: networkEndpoint.IP,
 			Port:      int64(portNum),
 		}
+
+		if networkEndpointType == negtypes.NonGCPPrivateEndpointType {
```
Comment (cxhiano) [inline pkg/neg/syncers/utils.go]: remove this. no need

File: pkg/neg/syncers/utils.go [hunk 4]
```diff
@@ -113,7 +112,7 @@ func getService(serviceLister cache.Indexer, namespace, name string) *apiv1.Serv
 }
 
 // ensureNetworkEndpointGroup ensures corresponding NEG is configured correctly in the specified zone.
-func ensureNetworkEndpointGroup(svcNamespace, svcName, negName, zone, negServicePortName string, cloud negtypes.NetworkEndpointGroupCloud, serviceLister cache.Indexer, recorder record.EventRecorder) error {
+func ensureNetworkEndpointGroup(svcNamespace, svcName, negName, zone, negServicePortName string, networkEndpointType negtypes.NetworkEndpointType, cloud negtypes.NetworkEndpointGroupCloud, serviceLister cache.Indexer, recorder record.EventRecorder) error {
```
Comment (freehan) [inline pkg/neg/syncers/utils.go]: Consider 2 things:

1. refactor the code to structure like this:
A. construct expected NEG resource
B. Check if the existing NEG is the same as expected
C. Delete/Create/No-Op 

2. Or add a TODO on top and leave it later.

**Pull request merged**